src/de/jost_net/JVerein/server/BuchungsartImpl.java

```diff
@@ -160,7 +156,14 @@ public void setStatus(int status) throws RemoteException
   @Override
   public Buchungsklasse getBuchungsklasse() throws RemoteException
   {
-    return (Buchungsklasse) getAttribute("buchungsklasse");
+    Long l = (Long) super.getAttribute("buchungsklasse");
+    if (l == null)
+    {
+      return null; // Keine Buchungsklasse zugeordnet
+    }
+
+    Cache cache = Cache.get(Buchungsklasse.class, true);
+    return (Buchungsklasse) cache.get(l);
```

Comment:
Kann hier `cache` auch mal `null` sein?

Reply:
Ist das Caching in der Funktion `getBuchungsklasseId()` auch nützlich?

Reply:
Ich habe die Methode so kopiert wie es in allen anderen Klassen mit dieser Methode auch ausschaut.
Ich glaube das Caching wurde mal gemacht um das Query einiger Tabellen zu beschleunigen.

Reply:
Das Caching funktioniert so schon eine ganze Zeit bei JVerein. Es wird auch nur für 10 Sekunden im Cache behalten damit nicht ganz viele Anfragen nacheinander kommen. Außerdem werden Buchungsarten und Klassen nicht so oft geändert das es hier zu Problemen kommen würde.